File: python_sdk/src/reality_capture/specifications/vector_optimization.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class VectorOptimizationInputs(BaseModel):
    vectors: list[str] = Field(description="Reality data Ids of vectors to consolidate")


class VectorOptimizationFormat(Enum):
    GEO_JSON = "GeoJSON"
    FEATURE_DB = "FeatureDB"


class VectorOptimizationOptions(BaseModel):
    out_format: Optional[VectorOptimizationFormat] = Field(None, description="Output format for the conversion.",
                                                     alias="format")
    input_crs: Optional[str] = Field(None, description="CRS for the input data", alias="inputCrs")
    output_crs: Optional[str] = Field(None, description="CRS for the output data", alias="outputCrs")
    feature_class_display_name: Optional[str] = Field(None, description="Display class name",
                                                      alias="featureClassDisplayName")


class VectorOptimizationSpecificationsCreate(BaseModel):
    inputs: VectorOptimizationInputs = Field(description="Inputs")
    options: Optional[VectorOptimizationOptions] = Field(None, description="Options")


class VectorOptimizationSpecifications(BaseModel):
    inputs: VectorOptimizationInputs = Field(description="Inputs")
    outputs: str = Field(description="Reality Data id of the vector data or Feature DB index (fdb:)")
    options: Optional[VectorOptimizationOptions] = Field(None, description="Options")

File: typescript/packages/reality-capture/src/specifications/vector_optimization.ts
import { z } from "zod";

export const VectorOptimizationInputsSchema = z.object({
  vectors: z
    .array(z.string())
    .describe("Reality data Ids of vectors to consolidate"),
});
export type VectorOptimizationInputs = z.infer<
  typeof VectorOptimizationInputsSchema
>;

export enum VectorOptimizationFormat {
  GEO_JSON = "GeoJSON",
  FEATURE_DB = "FeatureDB",
}

export const VectorOptimizationOptionsSchema = z.object({
  outputFormat: z
    .nativeEnum(VectorOptimizationFormat)
    .describe("Output format for the conversion.")
    .optional(),
  inputCrs: z.string().describe("CRS for the input data").optional(),
  outputCrs: z.string().describe("CRS for the output data").optional(),
  featureClassDisplayName: z.string().describe("Display class name").optional(),
});
export type VectorOptimizationOptions = z.infer<
  typeof VectorOptimizationOptionsSchema
>;

export const VectorOptimizationSpecificationsCreateSchema = z.object({
  inputs: VectorOptimizationInputsSchema.describe("Inputs"),
  options: VectorOptimizationOptionsSchema.describe("Options").optional(),
});
export type VectorOptimizationSpecificationsCreate = z.infer<
  typeof VectorOptimizationSpecificationsCreateSchema
>;

export const VectorOptimizationSpecificationsSchema = z.object({
  inputs: VectorOptimizationInputsSchema.describe("Inputs"),
  output: z
    .string()
    .describe("Reality Data id of the vector data or Feature DB index (fdb:)"),
  options: VectorOptimizationOptionsSchema.describe("Options").optional(),
});
export type VectorOptimizationSpecifications = z.infer<
  typeof VectorOptimizationSpecificationsSchema
>;
